service/storage_service.cc

```diff
@@ -2248,21 +2261,15 @@ void storage_service::run_bootstrap_ops(std::unordered_set<token>& bootstrap_tok
     node_ops_ctl ctl(*this, node_ops_cmd::bootstrap_prepare, db.get_config().host_id, get_broadcast_address());
     auto stop_ctl = deferred_stop(ctl);
     const auto& uuid = ctl.uuid();
+
+    // Step 1: Decide who needs to sync data for bootstrap operation
     // TODO: Specify ignore_nodes
+    ctl.start("bootstrap", [this_node = get_broadcast_address()] (gms::inet_address node) {
```

Comment:
We do not need to exclude this_node. Original code excludes this_node because we do not bother to check the gossip status of this_node.  In ctrl.start, get_endpoint_to_host_id_map_for_reading will not return this_node any way. 

Note, at line 2274, we insert this_node to sync_nodes explicitly. 

It is better to drop the this_node check filter here to avoid confusion.

Reply:
okay, makes sense

Reply:
Actually, it turns out that get_endpoint_to_host_id_map_for_reading does return pending nodes.
It's not restricted to normal town owners.
I'll delete the first patch then.
I'm tempted to change refresh_sync_nodes to return only normal token owners
but we should do that as a followup since it would impact decommission and removenode.